### DISCUSSION TITLE
PlayerService as Singleton for Blazor Server?

### DISCUSSION THREAD

**Discussion by leandrocervant** [microsoft/dotnet-podcasts#12] [Q&A]
Since the DI provides singleton services for webapplication instance and the PlayerService has episode interactions by user, is correct to register that as singleton?

wouldn't it be better to register as scoped?

https://github.com/microsoft/dotnet-podcasts/blob/42547f89071a8f17041aaaaa20b82de5b3f9c29c/src/Web/Server/Program.cs#L22

**Comment by behdad088** [comment]
Hi @leandrocervant  Looking at the class and what it does we dont need to create a new instance of that for each request!
